frontend/packages/console-app/src/components/user-preferences/utils/getTabContents.tsx

```diff
@@ -0,0 +1,50 @@
+import * as React from 'react';
+import { FormGroup } from '@patternfly/react-core';
+import { UserPreferencesInputType } from '@console/dynamic-plugin-sdk';
+import { ResolvedUserSettings } from '../types';
+import UserPreferenceInputCheckbox from '../UserPreferenceInputCheckbox';
+import UserPreferenceInputDropdown from '../UserPreferenceInputDropdown';
+
+export const createContentComponent = (userSettingsExtension: ResolvedUserSettings) => {
+  const { id, label, inputOption, customComponent, description } = userSettingsExtension;
+
+  if (customComponent && React.isValidElement(customComponent)) {
+    return (
+      <FormGroup key={id} fieldId={id} label={label} helperText={description} data-test={id}>
+        {customComponent}
+      </FormGroup>
+    );
+  }
+
+  if (inputOption.type === UserPreferencesInputType.dropdown) {
+    return (
+      <FormGroup key={id} fieldId={id} label={label} helperText={description} data-test={id}>
+        <UserPreferenceInputDropdown
+          id={id}
+          label={label}
+          userSettingKey={id}
+          dropdownOptions={inputOption.options}
+        />
+      </FormGroup>
+    );
+  }
```

Comment:
You should be able to create a map of type => component and then if present, simply render it without having to a big `if` condition per type.